dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -145,9 +146,16 @@ def handle_untrusted_email_signin(user)
     else
       # Otherwise, it's either the first login, or a user who must connect -
       # offer to connect the Clever account to an existing one, or insist if needed
-      session['clever_link_flag'] = user.provider
-      session['clever_takeover_id'] = user.uid
-      session['clever_takeover_token'] = user.oauth_token
+      if user.migrated?
+        auth_option = authentication_options.find_by credential_type: provider
```

Comment:
Where does `authentication_options` come from here?  Should it be `user.authentication_options.find_by ...`?

Reply:
Ick...this is why I really need tests here, you're right, this doesn't work. Adding some.